### ISSUE TITLE
Please write other replaygain tags to file

### ISSUE THREAD

**Issue by MrSimmo** [complexlogic/loudgain#141]
Would it be possible to add the ability to add other replaygain tags to the output file(s)?

For example:

Tag                                                                     Value
REPLAYGAIN_ALGORITHM                              ITU-R BS.1770
REPLAYGAIN_REFERENCE_LOUDNESS         -23 LUFS

This would show someone (or a player) what the replaygain values are associated with and the target output levels.

**Comment by thenktor** [comment]
I think this is a really good idea to have this information available. At least `loudgain` adds REPLAYGAIN_REFERENCE_LOUDNESS tags.
I'd also add the "Reference" to the output csv.

**Comment by DrZAjV** [comment]
Before this, someone mentioned it. The developers, considering compatibility, will not add irrelevant information to the audio files. Please refer to the previous post.

**Comment by thenktor** [comment]
This is not irrelevant information. If you are not using the default target loudness you really want to know which target loudness was used on the files.
Such an information also would be needed for a player to play files from different sources with different target loudness at the same volume level.
Furthermore even in the Musicbrainz Picard plugin there is a checkbox to add the REPLAYGAIN_REFERENCE_LOUDNESS tags.

**Comment by audiomuze** [comment]
Agree adding at least loudness reference would be good, adding the algorithm used as well even better.  Whilst not relevant to playback software, it's absolutely relevant to users wanting to ensure their entire library has been consistently analysed and doesn't have a blend of legacy and current replaygain scores.  Pretty obvious really.

**Comment by github8637** [comment]
> it's absolutely relevant to users wanting to ensure their entire library has been consistently analysed and doesn't have a blend of legacy and current replaygain scores. Pretty obvious really.

Pretty obvious such users would re-evaluate their entire library to a new algorithm, then just re-gain any additions...


> Such an information also would be needed for a player to play files from different sources with different target loudness at the same volume level. 

I do not know any players which use anything but standard R128 and REPLAYGAIN tags. Do you?


Furthermore you can just add any desired tag to files which were selected for the normalisation.

**Comment by audiomuze** [comment]
> Pretty obvious such users would re-evaluate their entire library to a new algorithm, then just re-gain any additions...


Yes, and then come back at it down the road not having a clue which algorithm was applied.

**Comment by Br31zh** [comment]
> I do not know any players which use anything but standard R128 and REPLAYGAIN tags. Do you?

Not yet, but I really hope it start to exist soon, because that be a really good way of getting rid of the debate about -23 (EBU R128 standard, Opus) vs. -18 LUFS (ReplayGain 2.0, despite being based on the EBU R128 standard).

I mean, rsgain even provide a `ebur128.ini` preset, but it can’t be used because players assume that Opus is -23 and everything else is -18. We can’t choose to follow the R128 standard on anything else than Opus…

**Comment by nift4** [comment]
gstreamer-based players such as Amarok use reference level tag and compensate to reach -18:
https://gitlab.freedesktop.org/gstreamer/gst-plugins-good/-/blob/1.16/gst/replaygain/gstrgvolume.c#L510

**Comment by Br31zh** [comment]
> gstreamer-based players such as Amarok use reference level tag and compensate to reach -18: https://gitlab.freedesktop.org/gstreamer/gst-plugins-good/-/blob/1.16/gst/replaygain/gstrgvolume.c#L510

I think it’s the case that was mentionned in another issue, gstreamer currently have a bug about this: https://gitlab.freedesktop.org/gstreamer/gstreamer/-/issues/2821 . But it’s on the way to be fixed: https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/9018

Which means that one of the example used to not implement this will soon be a reason to do it.
